exporter/collector/metricsexporter.go

```diff
@@ -878,16 +888,31 @@ func (m *metricMapper) labelDescriptors(pm pdata.Metric) []*label.LabelDescripto
 }
 
 // Returns (sum, count, percentile) metric names for a summary metric.
-func summaryMetricNames(name string) (string, string, string) {
+func (m *metricMapper) summaryMetricNames(name string) (string, string, string) {
+	if m.cfg.MetricConfig.SummaryPercentiles {
+		sumName := fmt.Sprintf("%s%s", name, LegacySummarySumSuffix)
+		countName := fmt.Sprintf("%s%s", name, LegacySummaryCountPrefix)
+		percentileName := fmt.Sprintf("%s%s", name, LegacySummaryPercentileSuffix)
+		return sumName, countName, percentileName
+	}
 	sumName := fmt.Sprintf("%s%s", name, SummarySumSuffix)
 	countName := fmt.Sprintf("%s%s", name, SummaryCountPrefix)
-	percentileName := fmt.Sprintf("%s%s", name, SummaryPercentileSuffix)
-	return sumName, countName, percentileName
+	return sumName, countName, name
 }
 
 func (m *metricMapper) summaryMetricDescriptors(pm pdata.Metric) []*metricpb.MetricDescriptor {
-	sumName, countName, percentileName := summaryMetricNames(pm.Name())
+	sumName, countName, percentileName := m.summaryMetricNames(pm.Name())
 	labels := m.labelDescriptors(pm)
+	quantileDescriptor := &label.LabelDescriptor{
+		Key:         "quantile",
+		Description: "the value at a given quantile of a distribution",
+	}
+	if m.cfg.MetricConfig.SummaryPercentiles {
+		quantileDescriptor = &label.LabelDescriptor{
```

Comment:
Nit: while it's clear enough in the context of this PR, I think a future reader would be confused that `quantileDescriptor` is actually about the percentile metric.

Reply:
renamed